apps/api/src/app/step-schemas/step-schemas.controller.ts

```diff
@@ -30,16 +34,56 @@ export class StepSchemasController {
   async getStepSchema(
     @UserSession() user: UserSessionData,
     @Param('stepType') stepType: StepType
-  ): Promise<SchemaTypeDto> {
-    const schema = await this.getStepDefaultSchemaUsecase.execute(
-      GetStepSchemaCommand.create({
+  ): Promise<StepSchemaDto> {
+    return await this.getStepDefaultSchemaUsecase.execute(
+      GetStepTypeSchemaCommand.create({
         organizationId: user.organizationId,
         environmentId: user.environmentId,
         userId: user._id,
         stepType,
       })
     );
+  }
+
+  @Get()
+  @ApiOperation({
+    summary: 'Get step schema for a specific workflow and step',
+    description: 'Get the schema for a specific step in a workflow',
+  })
+  @ApiParam({
+    name: 'workflowId',
+    type: String,
+    description: 'The ID of the workflow.',
+  })
+  @ApiParam({
+    name: 'stepId',
+    type: String,
+    description: 'The ID of the step within the workflow.',
+  })
+  @ExternalApiAccessible()
+  async getWorkflowStepSchema(
+    @UserSession() user: UserSessionData,
+    @Query('stepType') stepType: StepType,
+    @Query('workflowId') workflowId: string,
+    @Query('stepId') stepId: string
+  ): Promise<StepSchemaDto> {
+    return await this.getStepDefaultSchemaUsecase.execute(this.createCommand(user, stepType, workflowId, stepId));
+  }
 
-    return { schema };
+  private createCommand(user: UserSessionData, stepType: StepType, workflowId: string, stepId: string) {
+    return stepType
```

Comment:
self note: change priority it should be first by identifiers (workflowId and stepId) and then fallback to stepType